lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -84,6 +84,8 @@ ParseSILModuleRequest::evaluate(Evaluator &evaluator,
   if (hadError) {
     // The rest of the SIL pipeline expects well-formed SIL, so if we encounter
     // a parsing error, just return an empty SIL module.
+    assert(SF->getASTContext().hadError() &&
```

Comment:
I am not sure if we want this. I think what we want here is to error if we hadError and did not emit any diagnostics.

Reply:
where hadError is the thing returned by the SILParser.

Reply:
> I think what we want here is to error if we hadError and did not emit any diagnostics.

That’s what this condition does.

Reply:
If you haven’t emitted any errors via the diagnostic engine, `ASTContext::hadError` is false.

Reply:
True. Sorry wasn't thinking. The main thing that is important here is that we make it really obvious what we are doing here. With that in mind, I would suggest making it clear that what we are talking about is the SILParser failing without any /diagnostic/ emitted. I also would add that to the commit msg that we are specifically checking for that condition. Something like:

"We discovered that the SIL parser has certain fail code paths where we are not emitting diagnostics. This causes us to silently fail and just emit an empty SIL module. Detect this in +assert builds so that we can tell that a failure happened and to help in the future track down these missing diagnostics."